### DISCUSSION TITLE
Best way to access to doc.__dict__['name'] ?

### DISCUSSION THREAD

**Discussion by ovidioruzzier** [pymupdf/PyMuPDF#1848] [Looking for help] [answered]
#newbie#
I pass a doc to a procedure of mine and within that procedure I need to get/extract the filename the doc got the content from.
I don't think I must access to that variable in the way shown in the subject, do I?
#/newbie#

I found many answers about the descriptor, but I wonder if the fitz module provides a way to get that content (file's name)

Ovidio

**Comment by JorjMcKie** [comment, marked as answer]
I am not sure I understand your question, but you seem to mean the document's filename in your operating system. This is property `doc.name`.

**Reply by ovidioruzzier** [reply]
Thank you @JorjMcKie,

I went through dir(doc) but I oversaw the attribute 'name'.

Thanks again,

Ovidio
